### PR TITLE
executor: use batch get to speed up LOAD DATA

### PR DESCRIPTION
**Pull request by jackysp**
For loading a 4M rows file, this pr:
```
$ time mysql -uroot -P4000 -h127.0.0.1 -D test -e "load data local infile 'output.csv' into table sbtest1;" --local-infile

real	2m30.019s
user	0m0.027s
sys	0m0.091s
```
master branch:
```
$ time mysql -uroot -P5000 -h127.0.0.1 -D test -e "load data local infile 'output.csv' into table sbtest1;" --local-infile

real	33m21.829s
user	0m0.030s
sys	0m0.131s
```

### CODE REVIEW

**Comment by shenli** [comment]
Any benchmark result?

**Comment by jackysp** [comment]
PTAL @zimulala @winkyao

**Comment by ngaut** [comment]
Cool

**Comment by shenli** [comment]
@jackysp Please resolve the conflicts.

File: executor/write.go [hunk 1]
```diff
@@ -1007,9 +1022,21 @@ func (e *InsertExec) getKeysNeedCheck(rows [][]types.Datum) ([][]keyWithDupError
 // batchMarkDupRows marks rows with duplicate errors as nil.
 // All duplicate rows were marked and appended as duplicate warnings
 // to the statement context in batch.
-func (e *InsertExec) batchMarkDupRows(rows [][]types.Datum) ([][]types.Datum, error) {
+func batchMarkDupRows(i interface{}, rows [][]types.Datum) ([][]types.Datum, error) {
```
Comment (coocood) [inline executor/write.go]: I think pass `ctx` and `t` instead of interface{} is better.

Comment (jackysp) [reply executor/write.go]: I used to add a limitation for this two types. Anyway fixed. PTAL @coocood

**Comment by shenli** [comment]
LGTM

**Comment by shenli** [comment]
@coocood @winkyao PTAL

**Comment by coocood** [comment]
LGTM

**Pull request merged**